dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/validation/validator/event/DataRelationsValidatorTest.java

```diff
@@ -750,6 +757,113 @@ void eventValidationFailsWhenEventAOCAndEventCOsAreSetAndInProgramCCButNotAllCOs
                     && r.getMessage().contains(aoc.getUid())));
   }
 
+  @Test
+  void eventOfProgramWithRegistrationInEnrollmentWithoutTrackedEntity() {
+    OrganisationUnit orgUnit = organisationUnit(ORG_UNIT_ID);
+    when(preheat.getOrganisationUnit(MetadataIdentifier.ofUid(ORG_UNIT_ID))).thenReturn(orgUnit);
+
+    Program program = setupProgram(orgUnit);
+
+    CategoryCombo defaultCC = defaultCategoryCombo();
```

Comment:
the setups of these tests seems to be quite similar when it comes to category combos, could we have a helper method initialize it?